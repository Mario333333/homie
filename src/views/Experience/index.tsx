import briefcase from "../../assets/svg/briefcase.svg";
import ellipse from "../../assets/svg/ellipse.svg";
import InformationElement from "../../components/InformationElement";
import useAxios from "../../hooks/useAxios";
import { InformationElementData } from "../../utils/ts/componentTypes";

const Experience = () => {
  const { data, error } = useAxios<Array<InformationElementData>>("jobs");

  return (
    <div className="experience-container">
      {error === false &&
        data !== null &&
        data.map((item: InformationElementData, index: number) => {
          return (
            <InformationElement
              icon={index > 0 ? ellipse : briefcase}
              item={item}
              type={"job"}
              key={`job-${item.id}`}
            />
          );
        })}
    </div>
  );
};

export default Experience;
