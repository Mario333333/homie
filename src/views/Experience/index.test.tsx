import Experience from "./index";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from "@testing-library/react";
import { Router, Route, Switch } from "react-router-dom";
import { createMemoryHistory } from "history";
import Context from "../../app_context";

const history = createMemoryHistory();

describe("Experience", () => {
  afterEach(cleanup);
  const labels = {
    about_me: "Acerca de mi",
    contact: "Contacto",
    education: "EducaciÃ³n",
    experience: "Experiencia",
    portfolio: "Portafolio",
    knowledge_skills: "Conocimientos y habilidades",
    languages: "Idiomas",
    skills: "Habilidades",
    licenses_certifications: "Licencias y certificaciones",
    interests: "Intereses",
  };

  test("experience snapshot", () => {
    const { asFragment } = render(<Experience />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("experience defined component", () => {
    history.push("/experience");
    const {} = render(
      <Context.Provider value={{ labels }}>
        <Router history={history}>
          <Switch>
            <Route exact path="/experience">
              <Experience />
            </Route>
          </Switch>
        </Router>
      </Context.Provider>
    );
  });
});
