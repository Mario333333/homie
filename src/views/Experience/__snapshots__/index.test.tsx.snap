// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience experience snapshot 1`] = `
<DocumentFragment>
  <div
    class="experience-container"
  />
</DocumentFragment>
`;
